DataConnectors/CohesitySecurity/Helios2Sentinel/IncidentProducer/IncidentProducer.cs

```diff
@@ -0,0 +1,215 @@
+            dynamic alert, ILogger log)
+        {
+            dynamic output = new ExpandoObject();
+            output.properties = new ExpandoObject();
+            output.properties.Description = alert.alertDocument.alertDescription + ". Alert cause: " + alert.alertDocument.alertCause + ". Additional Info: " + alert.alertDocument.alertHelpText + ". Helios ID: " + alert.id;
+            output.properties.title = "Cluster: " + alert.clusterName;
+            output.properties.status = ((string)alert.alertState).Equals("kOpen", StringComparison.OrdinalIgnoreCase) ? "New" : "Closed";
+            output.properties.severity = "Medium";
+
+            string id = alert.id;
+            int i = 0;
+
+            foreach (var prop in alert.propertyList)
+            {
+                switch (i)
+                {
+                case 12:
+                    long sev = long.Parse((string)prop.value);
+                    if (sev >= 70)
+                        output.properties.severity = "High";
+                    else if (sev < 30)
+                        output.properties.severity = "Low";
+                    else
+                        output.properties.severity = "Medium";
+                    break;
+                case 0:
+                case 10:
+                case 4:
+                case 7:
+                case 11:
+                    TextWriter(id, (string)prop.key, (string)prop.value, log);
+                    break;
+                case 1:
+                    output.properties.title += ". Object: " + prop.value;
+                    TextWriter(id, (string)prop.key, (string)prop.value, log);
+                    break;
+                case 3:
+                    output.properties.title += ". Source: " + prop.value;
+                    break;
+                }
+                i++;
+                if (i == 13) break;
+            }
+
+            lock (queueLock)
+            {
+                outputQueueItem.Add(JsonConvert.SerializeObject(output));
+            }
+        }
+
+        private static void WriteData(string storageConnectionString, string containerName, string path, string data)
+        {
+            if (CloudStorageAccount.TryParse(storageConnectionString, out var storageAccount))
+            {
+                // Create the CloudBlobClient that represents the Blob storage endpoint for the storage account.
+                var cloudBlobClient = storageAccount.CreateCloudBlobClient();
+                var container = cloudBlobClient.GetContainerReference(containerName);
+                var blobRef = container.GetBlockBlobReference(path);
+                var dataAsBytes = Encoding.UTF8.GetBytes(data);
+                blobRef.UploadFromByteArrayAsync(dataAsBytes, 0, dataAsBytes.Length).Wait();
+            }
+            else
+            {
+                throw new InvalidProgramException("Invalid format string");
```

Comment:
Make the error message clearer and more specific, e.g. "Invalid storage connection string."